internal/internal_nexus_task_handler.go

```diff
@@ -0,0 +1,353 @@
+		taskQueueName:  taskQueueName,
+		client:         client,
+		metricsHandler: metricsHandler,
+		baseCompletion: baseCompletion,
+		baseFailure:    baseFailure,
+	}
+}
+
+func (h *nexusTaskHandler) Execute(task *workflowservice.PollNexusTaskQueueResponse) (*workflowservice.RespondNexusTaskCompletedRequest, *workflowservice.RespondNexusTaskFailedRequest, error) {
+	res, handlerErr, err := h.execute(task)
+	if err != nil {
+		return nil, nil, err
+	}
+	if handlerErr != nil {
+		return nil, h.fillInFailure(task.TaskToken, handlerErr), nil
+	}
+	return h.fillInCompletion(task.TaskToken, res), nil, nil
+}
+
+func (h *nexusTaskHandler) execute(task *workflowservice.PollNexusTaskQueueResponse) (*nexuspb.Response, *nexuspb.HandlerError, error) {
+	log, handlerErr := h.loggerForTask(task)
+	if handlerErr != nil {
+		return nil, handlerErr, nil
+	}
+	nctx := &NexusOperationContext{
+		Client:    h.client,
+		TaskQueue: h.taskQueueName,
+		Log:       log,
+	}
+	header := nexus.Header(task.GetRequest().GetHeader())
+	if header == nil {
+		header = nexus.Header{}
+	}
+
+	ctx, cancel, handlerErr := h.goContextForTask(nctx, header)
+	if handlerErr != nil {
+		return nil, handlerErr, nil
+	}
+	defer cancel()
+
+	switch req := task.GetRequest().GetVariant().(type) {
+	case *nexuspb.Request_StartOperation:
+		return h.handleStartOperation(ctx, nctx, req.StartOperation, header)
+	case *nexuspb.Request_CancelOperation:
+		return h.handleCancelOperation(ctx, nctx, req.CancelOperation, header)
+	default:
+		return nil, nexusHandlerError(nexus.HandlerErrorTypeNotImplemented, "unknown request type"), nil
+	}
+}
+
+func (h *nexusTaskHandler) handleStartOperation(ctx context.Context, nctx *NexusOperationContext, req *nexuspb.StartOperationRequest, header nexus.Header) (*nexuspb.Response, *nexuspb.HandlerError, error) {
+	serializer := &payloadSerializer{
+		converter: h.dataConverter,
+		payload:   req.GetPayload(),
+	}
+	// Create a fake lazy value, Temporal server already converts Nexus content into payloads.
+	input := nexus.NewLazyValue(
+		serializer,
+		&nexus.Reader{
+			ReadCloser: nopReadCloser,
+		},
+	)
+	// Ensure we don't pass nil values to handlers.
+	callbackHeader := req.GetCallbackHeader()
+	if callbackHeader == nil {
+		callbackHeader = make(map[string]string)
+	}
+	startOptions := nexus.StartOperationOptions{
+		RequestID:      req.RequestId,
+		CallbackURL:    req.Callback,
+		Header:         header,
+		CallbackHeader: callbackHeader,
+	}
+	var opres nexus.HandlerStartOperationResult[any]
+	var err error
+	func() {
+		defer func() {
+			recovered := recover()
+			if recovered != nil {
+				var ok bool
+				err, ok = recovered.(error)
+				if !ok {
+					err = fmt.Errorf("panic: %v", recovered)
+				}
+
+				nctx.Log.Error("Panic captured while handling nexus task", tagStackTrace, string(debug.Stack()), tagError, err)
+			}
+		}()
+		opres, err = h.nexusHandler.StartOperation(ctx, req.GetService(), req.GetOperation(), input, startOptions)
```

Comment:
I assume (inbound) interceptors are coming later?

Reply:
Yeah, I didn't include those for now.
I had them in a local branch but decided to start slim.

I'll make sure its tracked.